test/e2e/must_gather_operator_test.go

```diff
@@ -0,0 +1,759 @@
+				"Pod should still exist when RetainResourcesOnCompletion is true")
+			ginkgo.GinkgoWriter.Printf("Pod %s is retained after completion\n", retainedPodList.Items[0].Name)
+
+			ginkgo.GinkgoWriter.Println("RetainResourcesOnCompletion=true verified: Job and Pod are retained after completion")
+		})
+	})
+
+	ginkgo.Context("Security and Isolation Tests", func() {
+		var mg *mustgatherv1alpha1.MustGather
+
+		ginkgo.AfterEach(func() {
+			if mg != nil {
+				ginkgo.By("Cleaning up MustGather CR")
+				_ = nonAdminClient.Delete(testCtx, mg)
+				mg = nil
+			}
+		})
+
+		ginkgo.It("should fail when non-admin user tries to use privileged ServiceAccount", func() {
+			mustGatherName := fmt.Sprintf("test-privileged-sa-%d", time.Now().UnixNano())
+
+			ginkgo.By("Attempting to create MustGather with non-existent privileged SA")
+			mg = createMustGatherCR(mustGatherName, ns.Name, "cluster-admin-sa", false) // SA doesn't exist
+
+			ginkgo.By("Verifying Job is created but Pod fails due to missing SA")
+			job := &batchv1.Job{}
+			Eventually(func() error {
+				return adminClient.Get(testCtx, client.ObjectKey{
+					Name:      mustGatherName,
+					Namespace: ns.Name,
+				}, job)
+			}).WithTimeout(2 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+
+			// Job should exist but pods won't be created due to missing SA
+			Consistently(func() int32 {
+				_ = adminClient.Get(testCtx, client.ObjectKey{
+					Name:      mustGatherName,
+					Namespace: ns.Name,
+				}, job)
+				return job.Status.Active
+			}).WithTimeout(30*time.Second).WithPolling(5*time.Second).Should(Equal(int32(0)),
+				"Job should not have active pods due to missing ServiceAccount")
+		})
+
+		ginkgo.It("should enforce ServiceAccount permissions during data collection", func() {
+			mustGatherName := fmt.Sprintf("test-sa-permissions-%d", time.Now().UnixNano())
+
+			ginkgo.By("Creating MustGather with SA")
+			mg = createMustGatherCR(mustGatherName, ns.Name, serviceAccount, false)
+
+			ginkgo.By("Waiting for Pod to start")
+			var gatherPod *corev1.Pod
+			Eventually(func() bool {
+				podList := &corev1.PodList{}
+				if err := adminClient.List(testCtx, podList,
+					client.InNamespace(ns.Name),
+					client.MatchingLabels{jobNameLabelKey: mustGatherName}); err != nil {
+					return false
+				}
+				if len(podList.Items) > 0 {
+					gatherPod = &podList.Items[0]
+					return true
+				}
+				return false
+			}).WithTimeout(2 * time.Minute).WithPolling(5 * time.Second).Should(BeTrue())
+
+			ginkgo.By("Verifying Pod has no privilege escalation")
+			Expect(gatherPod.Spec.ServiceAccountName).To(Equal(serviceAccount))
+
+			// Check that pod doesn't request privileged mode
+			for _, container := range gatherPod.Spec.Containers {
+				if container.SecurityContext != nil && container.SecurityContext.Privileged != nil {
+					Expect(*container.SecurityContext.Privileged).To(BeFalse(),
+						"Container should not run in privileged mode")
+				}
+			}
```

Comment:
this is sufficient check, I'd say.
Or if you want to take further you can check the SCC/PSA annotations on the pod not being "privileged".